api/python/tests/integration/test_packages.py

```diff
@@ -666,6 +672,70 @@ def test_s3_set_dir(self):
 
             list_object_versions_mock.assert_called_with('bucket', 'foo/')
 
+    def test_set_dir_update_policy(self):
+        """Verify building a package with update policy. """
+
+        # Test for local src dir
+        path = pathlib.Path(__file__).parent
+        data_dir = pathlib.Path(path, "data")
+        pkg = Package()
+        pkg = pkg.set_dir("/", data_dir, meta={'name': 'test_meta'})
+        assert 'nested' in pkg.keys()
+        assert pkg.meta == {'name': 'test_meta'}
+        existing_one_entry = pkg['nested/one.txt']
+
+        nested_dir = pathlib.Path(data_dir, "nested2")
+        nested_dir.mkdir(parents=True, exist_ok=True)
+        with open(nested_dir / 'one.txt', 'w') as fd:
+            fd.write(fd.name)
+
+        # existing update policy
```

Comment:
it's cleaner to use pytest.parameterize and make for multiple invocations of this unit test function with different parameters: one with "existing", one with "incoming", etc. See https://github.com/quiltdata/quilt/blob/master/lambdas/thumbnail/tests/test_thumbnail.py#L46. This avoids side-effects, is closer to how things are called in the real world, etc.